src/utils/diff/parse.ts

```diff
@@ -4,7 +4,12 @@ import { AndroidParser, CsharpParser, GolangParser, IosParser, JavaParser,
 import { ParseOptions, VariableMatch } from './parsers/types'
 import { CustomParser } from './parsers/custom'
 
-const PARSERS: Record<string, (typeof NodeParser)[]> = {
+type Parser = typeof CustomParser | typeof NodeParser | typeof ReactParser | typeof JavascriptParser
```

Comment:
do you need to do this? from the perspective of this file the parsers should all look the same (which is why I just chose one to represent the type)

Reply:
Since some of the properties are optional (for example orderedParameterPatterns), `typeof IosParser` and `typeof NodeParser` are different

Reply:
shouldn't those be "protected" then? The actual external signature of all of these should be the same